### PR TITLE
[FLINK-15151][table-planner-blink] Use new type system in TableSource…

### PR DESCRIPTION
**Pull request by docete**
…Util.computeIndexMapping of blink planner

## What is the purpose of the change

`TableSourceUtil.computeIndexMapping` used to compute indices that map the input type of the datastream to the schema of the table.  Since we introduced new type system, we should use it to avoid conversion between DataType and TypeInformation

## Brief change log

- fbbc9e5 Use new type system in `TableSourceUtil.computeIndexMapping`

## Verifying this change

This change is already covered by new tests, such as *TableSourceITCase.testProjectablePoJoTable*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / docs / **JavaDocs** / not documented)

### CODE REVIEW

**Comment by rmetzger** [comment]
(Note, the azure failure is probably a build instability tracked here: https://issues.apache.org/jira/browse/FLINK-14592)

File: flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSourceITCase.scala [hunk 1]
```diff
@@ -329,4 +331,44 @@ class TableSourceITCase extends BatchTestBase {
       )
     )
   }
+
+  @Test
+  def testProjectablePoJoTable(): Unit = {
```
Comment (wuchong) [inline flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSourceITCase.scala]: What's the purpose of this test? I revert the changes in `TableSourceUtil` and this test is still passed.

Comment (wuchong) [reply flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/runtime/batch/sql/TableSourceITCase.scala]: I think we should have a test that there is a DECIMAL type defined the source which may should use DDL.

**Comment by docete** [comment]
close this since it's superceded by FLINK-15124
